Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkBasic/PsychEyelinkDispatchCallback.m

```diff
@@ -129,326 +161,419 @@
-        else
-            % Calibration success:
-            EyelinkMakeSound(el, 'calibration_success_beep');
+        
+        if strcmpi(el.calTargetType, 'video') && ~isempty(eyelinkanimationtarget)
+             texkill=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTexClose);
+             Screen('PlayMovie', eyelinkanimationtarget.movie, 0, el.calAnimationLoopParam);
+             if texkill > 0
+                 Screen('Close', texkill);
+             end
+        end
+        
+    case 11 % Exit Cal Display
+        calxy = [];
+        if(~isempty(eyelinkanimationtarget)	)
+            eyelinkanimationtarget.calxy=calxy;
         end
-    case 15,
-        % Drift correction done sound:
-        errc = callArgs(2);
-        %         fprintf('dc_done_beep_hook: %i\n', errc);
-        if errc > 0
-            % Drift correction failed:
-            EyelinkMakeSound(el, 'drift_correction_failed_beep');
+        
+        if inDrift
+            inDrift = 0;
+            drawInstructions = 0;
         else
-            % Drift correction success:
-            EyelinkMakeSound(el, 'drift_correction_success_beep');
+            drawInstructions = 1;
+        end
+        
+        clearScreen=1;
+        needsupdate=1;
+        if strcmpi(el.calTargetType, 'video') && ~isempty(eyelinkanimationtarget)
+            texkill=Screen('GetMovieImage', eyewin, eyelinkanimationtarget.movie, el.calAnimationWaitTexClose);
+            Screen('PlayMovie', eyelinkanimationtarget.movie, 0, el.calAnimationLoopParam);
+            if texkill > 0
+                Screen('Close', texkill);
+            end
         end
-        % add by NJ
-    case 16,
+        
+    case 12 % New Cal Target Sound
+        if ~strcmpi(el.calTargetType, 'video')
+            EyelinkMakeSound(el, 'cal_target_beep');
+        end
+        
+    case 13 % New Drift Chk/Corr Sound
+        if ~strcmpi(el.calTargetType, 'video')
+            EyelinkMakeSound(el, 'drift_correction_target_beep');
+        end
+        
+    case 14 % Cal Done Sound
+        if ~strcmpi(el.calTargetType, 'video')
+            errc = callArgs(2);
+            if errc > 0
+                % Failed
+                EyelinkMakeSound(el, 'calibration_failed_beep');
+            else
+                % Success
+                EyelinkMakeSound(el, 'calibration_success_beep');
+            end
+        end
+        
+    case 15 % Drift Chk/Corr Done Sound
+        if ~strcmpi(el.calTargetType, 'video')
+            errc = callArgs(2);
+            if errc > 0
+                % Failed
+                EyelinkMakeSound(el, 'drift_correction_failed_beep');
+            else
+                % Success
+                EyelinkMakeSound(el, 'drift_correction_success_beep');
+            end
+        end
+        
+    case 16 % Get Mouse Position
         [width, height]=Screen('WindowSize', eyewin);
-        % get mouse
         [x,y, buttons] = GetMouse(eyewin);
```

Comment:
More of a suggestion for the future:
One could follow this with a ``[x,y] = RemapMouse(eyewin, viewid, x, y);`` for suitable viewid, depending on stereomode, if any. See `help RemapMouse` for the why.

This could be generally useful also to remap gaze positions, not only mouse pos, in case of geometric display correction active.

Reply:
will further explore this suggestion for a future revision